lib/chef_zero/server.rb

```diff
@@ -164,10 +164,10 @@ def port
     #
     def url
       sch = @options[:ssl] ? 'https' : 'http'
-      @url ||= if @options[:host].include?(':')
-                 URI("#{sch}://[#{@options[:host]}]:#{port}").to_s
+      @url ||= if @options[:host].first.include?(':')
+                 URI("#{sch}://[#{@options[:host].first}]:#{port}").to_s
                else
-                 URI("#{sch}://#{@options[:host]}:#{port}").to_s
+                 URI("#{sch}://#{@options[:host].first}:#{port}").to_s
```

Comment:
I forget: is this used when chef-zero generates URLs that it puts in documents it returns? In most cases we ignore those on the client side, but the cookbook upload/sandbox stuff does use them. For example, if you configure `::1` and `127.0.0.1` (in that order) and then try to upload cookbooks to `127.0.0.1`, would the server return IPv6 addresses for the checksums to upload?